### PR TITLE
Feat/civ 3773 SDO RPA additions

### PR DESCRIPTION
**Pull request by m-meulendijks-v1**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-3773

### Change description ###

Change to the RPA journey to include SDO.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java [hunk 1]
```diff
@@ -40,7 +40,8 @@ public enum CaseEvent {
     INITIATE_GENERAL_APPLICATION(USER),
     CREATE_SDO(USER),
     STANDARD_DIRECTION_ORDER_DJ(USER),
-
+    NotSuitable_SDO(USER),
```
Comment (mounikahmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java:43]: Why is it in lower case?

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -205,7 +205,7 @@ launchdarkly:
 
 testing:
   support:
-    enabled: ${TESTING_SUPPORT_ENABLED:false}
+    enabled: ${TESTING_SUPPORT_ENABLED:true}
```
Comment (mounikahmcts) [inline src/main/resources/application.yaml:208]: We are not supposed to enable it always. It should be enabled only if required.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/NotSuitableSDOCallbackHandler.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    private String getHeader(CaseData caseData) {
+        return format("# Your request was accepted%n## Case has now moved offline");
+    }
+
+    protected CallbackResponse emptyCallbackResponse(CallbackParams callbackParams) {
+        return AboutToStartOrSubmitCallbackResponse.builder().build();
+    }
+
+    private String getBody(CaseData caseData) {
+        return format(NotSuitableSDO_CONFIRMATION_BODY);
```
Comment (mounikahmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/NotSuitableSDOCallbackHandler.java:100]: Naming is not very consistent

**Review by mounikahmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java (hunk 1 above)
Comment (m-meulendijks-v1) [reply src/main/java/uk/gov/hmcts/reform/civil/callback/CaseEvent.java:43]: I am merely following how it was named in PR CIV-3328 - I had to merge their changes into my branch to make the RPA for their unhappy path work.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/NotSuitableSDOCallbackHandler.java (hunk 3 above)
Comment (m-meulendijks-v1) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/NotSuitableSDOCallbackHandler.java:100]: This class was merged in from the CIV-3328 branch. I only made the method 'addUnsuitableSDODate'.

File: src/main/resources/application.yaml (hunk 2 above)
Comment (m-meulendijks-v1) [reply src/main/resources/application.yaml]: Change made to disable it

**Review by sampankumar — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/FlowStateAllowedEventService.java [hunk 4]
```diff
@@ -426,18 +409,15 @@ public class FlowStateAllowedEventService {
...
+                CREATE_SDO
             )
         ),
         entry(
             PAST_CLAIM_DETAILS_NOTIFICATION_DEADLINE_AWAITING_CAMUNDA.fullName(),
             List.of(
                 DISMISS_CLAIM,
                 ADD_CASE_NOTE,
-                INITIATE_GENERAL_APPLICATION,
-                CREATE_SDO,
```
Comment (nagaraju-deekonda) [inline src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/FlowStateAllowedEventService.java:439]: why did you remove CREATE_SDO here

Comment (m-meulendijks-v1) [reply src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/FlowStateAllowedEventService.java:439]: To align with a unit test when I was rebasing - should it be there? I can put it back & adjust unit test.

**Review by nagaraju-deekonda — DISMISSED** [review]

**Review by mounikahmcts — APPROVED** [review]

**Review by sampankumar — APPROVED** [review]

**Review by cj1993 — APPROVED** [review]
